### ISSUE TITLE
Incorrect migration created for nested owned collection without model changes

### ISSUE THREAD

**Issue by J-Yen** [dotnet/efcore#23493]
When having a complex owned type hierarchy like reproduced in the sample below adding a new initial migration from scratch works as expected. However, when immediately adding a new migration while not changing anything on the model itself Entity Framework is generating a silly migration with the creation of the same tables.

1. Creation of the context:
```C#
using System;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore;

namespace EfSillyMigration
{
    class Program
    {
        static void Main(string[] args)
        {
            using var dbContext = new ExampleContext();
            Console.ReadLine();
        }

        public class ExampleContext : DbContext
        {
            protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
                => optionsBuilder.UseSqlServer("Server=(localdb)\\mssqllocaldb;Database=TestDb;Trusted_Connection=True;MultipleActiveResultSets=true");

            public DbSet<AnAggregateRoot> AggregateRoots { get; set; }

            protected override void OnModelCreating(ModelBuilder mb)
            {
                base.OnModelCreating(mb);
                mb.Entity<AnAggregateRoot>().HasKey(x => x.Id);
                mb.Entity<AnAggregateRoot>()
                    .OwnsOne(e => e.FirstOwnedDependent1, x => x.OwnsMany(e => e.OwnedDependents2, y =>
                    {
                        y.ToTable("FirstOwnedDependent2");
                        y.HasKey(e => e.Id);
                        y.OwnsMany(e => e.OwnedDependents3, z =>
                        {
                            z.ToTable("FirstOwnedDependent3");
                            z.HasKey(e => e.Id);
                        });
                    }))
                    .OwnsOne(e => e.SecondOwnedDependent1, x => x.OwnsMany(e => e.OwnedDependents2, y =>
                    {
                        y.ToTable("SecondOwnedDependent2");
                        y.HasKey(e => e.Id);
                        y.OwnsMany(e => e.OwnedDependents3, z =>
                        {
                            z.ToTable("SecondOwnedDependent3");
                            z.HasKey(e => e.Id);
                        });
                    }));
            }
        }
    }

    public class AnAggregateRoot
    {
        public int Id { get; set; }
        public string AProperty { get; set; }
        public OwnedDependent1 FirstOwnedDependent1 { get; set; }
        public OwnedDependent1 SecondOwnedDependent1 { get; set; }
    }

    public class OwnedDependent1
    {
        public string AProperty { get; set; }
        public ICollection<OwnedDependent2> OwnedDependents2 { get; set; }
    }

    public class OwnedDependent2 
    {
        public int Id { get; set; }
        public string AProperty { get; set; }
        public ICollection<OwnedDependent3> OwnedDependents3 { get; set; }
    }

    public class OwnedDependent3
    {
        public int Id { get; set; }
        public string AProperty { get; set; }
    }
}
```

2: Initial migration creates the expected tables and relationships.
```
Add-Migration Init
```
```C#
using Microsoft.EntityFrameworkCore.Migrations;

namespace EFSillyMigration.Migrations
{
    public partial class Init : Migration
    {
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.CreateTable(
                name: "AggregateRoots",
                columns: table => new
                {
                    Id = table.Column<int>(type: "int", nullable: false)
                        .Annotation("SqlServer:Identity", "1, 1"),
                    AProperty = table.Column<string>(type: "nvarchar(max)", nullable: true),
                    FirstOwnedDependent1_AProperty = table.Column<string>(type: "nvarchar(max)", nullable: true),
                    SecondOwnedDependent1_AProperty = table.Column<string>(type: "nvarchar(max)", nullable: true)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_AggregateRoots", x => x.Id);
                });

            migrationBuilder.CreateTable(
                name: "FirstOwnedDependent2",
                columns: table => new
                {
                    Id = table.Column<int>(type: "int", nullable: false)
                        .Annotation("SqlServer:Identity", "1, 1"),
                    AProperty = table.Column<string>(type: "nvarchar(max)", nullable: true),
                    OwnedDependent1AnAggregateRootId = table.Column<int>(type: "int", nullable: false)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_FirstOwnedDependent2", x => x.Id);
                    table.ForeignKey(
                        name: "FK_FirstOwnedDependent2_AggregateRoots_OwnedDependent1AnAggregateRootId",
                        column: x => x.OwnedDependent1AnAggregateRootId,
                        principalTable: "AggregateRoots",
                        principalColumn: "Id",
                        onDelete: ReferentialAction.Cascade);
                });

            migrationBuilder.CreateTable(
                name: "SecondOwnedDependent2",
                columns: table => new
                {
                    Id = table.Column<int>(type: "int", nullable: false)
                        .Annotation("SqlServer:Identity", "1, 1"),
                    AProperty = table.Column<string>(type: "nvarchar(max)", nullable: true),
                    OwnedDependent1AnAggregateRootId = table.Column<int>(type: "int", nullable: false)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_SecondOwnedDependent2", x => x.Id);
                    table.ForeignKey(
                        name: "FK_SecondOwnedDependent2_AggregateRoots_OwnedDependent1AnAggregateRootId",
                        column: x => x.OwnedDependent1AnAggregateRootId,
                        principalTable: "AggregateRoots",
                        principalColumn: "Id",
                        onDelete: ReferentialAction.Cascade);
                });

            migrationBuilder.CreateTable(
                name: "FirstOwnedDependent3",
                columns: table => new
                {
                    Id = table.Column<int>(type: "int", nullable: false)
                        .Annotation("SqlServer:Identity", "1, 1"),
                    AProperty = table.Column<string>(type: "nvarchar(max)", nullable: true),
                    OwnedDependent2Id = table.Column<int>(type: "int", nullable: false)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_FirstOwnedDependent3", x => x.Id);
                    table.ForeignKey(
                        name: "FK_FirstOwnedDependent3_FirstOwnedDependent2_OwnedDependent2Id",
                        column: x => x.OwnedDependent2Id,
                        principalTable: "FirstOwnedDependent2",
                        principalColumn: "Id",
                        onDelete: ReferentialAction.Cascade);
                });

            migrationBuilder.CreateTable(
                name: "SecondOwnedDependent3",
                columns: table => new
                {
                    Id = table.Column<int>(type: "int", nullable: false)
                        .Annotation("SqlServer:Identity", "1, 1"),
                    AProperty = table.Column<string>(type: "nvarchar(max)", nullable: true),
                    OwnedDependent2Id = table.Column<int>(type: "int", nullable: false)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_SecondOwnedDependent3", x => x.Id);
                    table.ForeignKey(
                        name: "FK_SecondOwnedDependent3_SecondOwnedDependent2_OwnedDependent2Id",
                        column: x => x.OwnedDependent2Id,
                        principalTable: "SecondOwnedDependent2",
                        principalColumn: "Id",
                        onDelete: ReferentialAction.Cascade);
                });

            migrationBuilder.CreateIndex(
                name: "IX_FirstOwnedDependent2_OwnedDependent1AnAggregateRootId",
                table: "FirstOwnedDependent2",
                column: "OwnedDependent1AnAggregateRootId");

            migrationBuilder.CreateIndex(
                name: "IX_FirstOwnedDependent3_OwnedDependent2Id",
                table: "FirstOwnedDependent3",
                column: "OwnedDependent2Id");

            migrationBuilder.CreateIndex(
                name: "IX_SecondOwnedDependent2_OwnedDependent1AnAggregateRootId",
                table: "SecondOwnedDependent2",
                column: "OwnedDependent1AnAggregateRootId");

            migrationBuilder.CreateIndex(
                name: "IX_SecondOwnedDependent3_OwnedDependent2Id",
                table: "SecondOwnedDependent3",
                column: "OwnedDependent2Id");
        }

        protected override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropTable(
                name: "FirstOwnedDependent3");

            migrationBuilder.DropTable(
                name: "SecondOwnedDependent3");

            migrationBuilder.DropTable(
                name: "FirstOwnedDependent2");

            migrationBuilder.DropTable(
                name: "SecondOwnedDependent2");

            migrationBuilder.DropTable(
                name: "AggregateRoots");
        }
    }
}
```
3: When immediately adding a new migration while not changing anything on the model I get unexpected table creation statements (exactly the same table creations).  
```
Add-Migration EmptyUpdate
```

```C#
using Microsoft.EntityFrameworkCore.Migrations;

namespace EFSillyMigration.Migrations
{
    public partial class Update : Migration
    {
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.CreateTable(
                name: "FirstOwnedDependent3",
                columns: table => new
                {
                    Id = table.Column<int>(type: "int", nullable: false)
                        .Annotation("SqlServer:Identity", "1, 1"),
                    AProperty = table.Column<string>(type: "nvarchar(max)", nullable: true),
                    OwnedDependent2Id = table.Column<int>(type: "int", nullable: false)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_FirstOwnedDependent3", x => x.Id);
                    table.ForeignKey(
                        name: "FK_FirstOwnedDependent3_FirstOwnedDependent2_OwnedDependent2Id",
                        column: x => x.OwnedDependent2Id,
                        principalTable: "FirstOwnedDependent2",
                        principalColumn: "Id",
                        onDelete: ReferentialAction.Cascade);
                });

            migrationBuilder.CreateIndex(
                name: "IX_FirstOwnedDependent3_OwnedDependent2Id",
                table: "FirstOwnedDependent3",
                column: "OwnedDependent2Id");
        }

        protected override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropTable(
                name: "FirstOwnedDependent3");
        }
    }
}

```

Any workarounds possible or bugfix possible?

EF Core version: 3.1.9 and 5.0.0
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.1 and .NET 5.0
Operating system: Windows 1909
IDE: Visual Studio 2019 16.7.6

**Comment by J-Yen** [comment]
Similar: #23496

**Comment by AndriySvyryd** [comment]
This was fixed in 6.0
